### PR TITLE
gimme-aws-creds: fix build override fido2 dependency

### PR DESCRIPTION
**Pull request by jbgosselin**
## Description of changes

With the bump of `fido2` dependency, the project toolchain is using pyproject. Additionally, dev dependencies have been changed. `fido2` prior to v1.0.0 needed a specific command to run test properly.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/admin/gimme-aws-creds/default.nix [hunk 1]
```diff
@@ -14,11 +14,22 @@ let
       fido2 = super.fido2.overridePythonAttrs (oldAttrs: rec {
         version = "0.9.3";
         format = "setuptools";
+        pyproject = null;
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/gimme-aws-creds/default.nix:17]:
```suggestion
```
Isn't this working?

Comment (jbgosselin) [reply pkgs/tools/admin/gimme-aws-creds/default.nix:17]: This returns the following error
```
error:
       … while evaluating the attribute 'drvPath'
         at /nix/store/dv1bg4hqjsf4mi21rncvmfhgbac5ckg1-source/lib/customisation.nix:365:7:
          364|     in commonAttrs // {
          365|       drvPath = assert condition; drv.drvPath;
             |       ^
          366|       outPath = assert condition; drv.outPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:9:12:
            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: assertion '((pyproject != null) -> (format == null))' failed
       at /nix/store/dv1bg4hqjsf4mi21rncvmfhgbac5ckg1-source/pkgs/development/interpreters/python/mk-python-derivation.nix:156:1:
          155|
          156| assert (pyproject != null) -> (format == null);
             | ^
          157|
```

Comment (SuperSandro2000) [reply pkgs/tools/admin/gimme-aws-creds/default.nix:17]: Can we try to keep pyproject = true?

File: pkgs/development/python-modules/ctap-keyring-device/default.nix [hunk 2]
```diff
@@ -5,28 +5,35 @@
...
   fido2_0 = fido2.overridePythonAttrs (oldAttrs: rec {
     version = "0.9.3";
-    format = "setuptools";
     src = fetchPypi {
       inherit (oldAttrs) pname;
       inherit version;
       hash = "sha256-tF6JphCc/Lfxu1E3dqotZAjpXEgi+DolORi5RAg0Zuw=";
     };
+    build-system = [ setuptools-scm ];
+    propagatedBuildInputs = [ six ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ctap-keyring-device/default.nix:27]: With the new pyptoject flag this should be dependencies

File: pkgs/development/python-modules/ctap-keyring-device/default.nix [hunk 3]
```diff
@@ -5,28 +5,35 @@
...
+    nativeCheckInputs = [ unittestCheckHook ];
+    checkInputs = [ mock ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ctap-keyring-device/default.nix:29]: And this goes into native check inputs, too

File: pkgs/development/python-modules/ctap-keyring-device/default.nix [hunk 4]
```diff
@@ -5,28 +5,35 @@
...
   });
 in
 buildPythonPackage rec {
   pname = "ctap-keyring-device";
   version = "1.0.6";
-  format = "setuptools";
+  pyproject = true;
+  build-system = [ setuptools-scm ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ctap-keyring-device/default.nix:36]: This should be between postPatch and nativeBuildInputs. It is basically another list of inputs.

File: pkgs/tools/admin/gimme-aws-creds/default.nix [hunk 5]
```diff
@@ -13,12 +13,15 @@ let
     packageOverrides = self: super: {
       fido2 = super.fido2.overridePythonAttrs (oldAttrs: rec {
         version = "0.9.3";
-        format = "setuptools";
         src = fetchPypi {
           inherit (oldAttrs) pname;
           inherit version;
           hash = "sha256-tF6JphCc/Lfxu1E3dqotZAjpXEgi+DolORi5RAg0Zuw=";
         };
+        build-system = [ super.setuptools-scm ];
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/gimme-aws-creds/default.nix:21]: Same as above

**Pull request merged**